src/nrniv/neuronapi.h

```diff
@@ -0,0 +1,111 @@
+#pragma once
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+typedef struct Symbol Symbol;
+typedef struct Object Object;
+typedef struct Section Section;
+typedef struct SectionListIterator SectionListIterator;
+typedef struct nrn_Item nrn_Item;
+typedef struct SymbolTableIterator SymbolTableIterator;
+typedef struct Symlist Symlist;
+
+typedef enum {
+    STACK_IS_STR = 1,
+    STACK_IS_VAR = 2,
+    STACK_IS_NUM = 3,
+    STACK_IS_OBJVAR = 4,
+    STACK_IS_OBJTMP = 5,
+    STACK_IS_USERINT = 6,
```

Comment:
@nrnhines is there a reason for the "userint" name? instead of int?

Reply:
I don't think there is a reason in the context of a stack type. I see that almost all appearance of USERINT is in the context of a Symbol subtype.  I see that when a Symbol on the stack is evaluated and it happens to be a USERINT subtype, what actually gets pushed back onto the stack is  a double
```
             case USERINT:
                d = (double) (*(sym->u.pvalint));
                break;
```
and assignment pops a double and casts to a int
```
*(sym->u.pvalint) = (int) (d2 + hoc_epsilon);
```
So I think it would be a good idea NOT to use USERINT for the stack type. But perhaps it suffices just to change the api so that
```
    STACK_IS_INT = 6,
```